helga_jeopardy.py

```diff
@@ -267,6 +361,170 @@ def scores(client, channel, nick, alltime=False):
+    control = random.choice(current_players)
+    client.msg(channel, "By random choice, {} has control of the board".format(control))
+    client.msg(channel, "To select a question, use the following format: !j cat3 200")
+    mongo_db.update({
+        'game_active': True,
+        'game_started': False,
+    }, {
+        '$set': {
+            'game_started': True,
+            'control': control,
+       }
+    })
+    reactor.callLater(1800, end_game, client, channel)
+    return
+
+def show_board(client, channel, mongo_db=db.jeopardy):
+    current_game = mongo_db.find_one({
+        'channel': channel,
+        'game_active': True,
+    })
+
+    for key, value in current_game.items():
+        if key.startswith('cat'):
+            values = [str(value['clue{}'.format(i)]['value']) for i in range(1, 6) if value['clue{}'.format(i)]['active']]
+            cat_names = value["category"]
+            score_response =' '.join([key,  cat_names, ' '.join(sorted(values, key=int))])
+            client.msg(channel, score_response)
+
+    players = current_game['players']
+    client.msg(channel, "Here are the current scores:")
+    for i in players:
+        client.msg(channel, "{} : {}".format(i, current_game[i]))
+
+def evaluate_control(client, channel, nick, current_game, sel_category, sel_value, quest_func=retrieve_question):
+    if current_game["control"] == nick:
+        result = quest_func(client, channel, current_game, sel_category, sel_value, random=False)
+        return result
+    else:
+        client.msg(channel, "You do not have control of the board")
+        return
+
+def check_remaining_clues(client, channel, mongo_db=db.jeopardy):
+    current_game = mongo_db.find_one({
+        'channel': channel,
+        'game_active': True,
+        'game_started': True,
+    })
+    question_count = 0
+    for key, value in current_game.items():
+        if key.startswith('cat'):
+            for k, v in value.items():
+                if k.startswith('clue'):
+                    if v['active'] == True:
+                        question_count += 1
+
+    if question_count < 1:
+        client.msg(channel, "All questions have been answered. Ending game..")
+        end_game(client, channel, current_game)
+        return False
+    else:
+        return True
+
+def end_game(client, channel, current_game=None, mongo_db=db.jeopardy):
+    mongo_db.update({
+	'game_active': True,
+	'channel': channel,
+    }, {
+	'$set': {
+	    'game_active': False,
+	    'game_started': False
+	}
+    })
+    if current_game:
+        current_players = current_game['players']
+        client.msg(channel, "Here are the final scores:")
+        scores = []
+        for i in current_players:
+	    client.msg(channel, "{} : {}".format(i, current_game[i]))
+            scores.append(current_game[i])
+        
+        winners = []
+        for key, value in current_game.items():
+            if value == max(scores):
+                winners.append(key)
```

Comment:
With players as dicts:

```suggestion
        scores = []
        winners = []
        max_score = max(player['score'] for player in current_game['players'])
        
        client.msg(channel, 'Here are the final scores:')
        for player in current_game['players']:
	    client.msg(channel, '{}: ${}'.format(player['nick'], player['score']))
	    if player['score'] == max_score:
                winners.append(player['nick'])
```